### ISSUE TITLE
[AXE-CORE]: GIBCT EYB - Form elements MUST have labels

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#14407]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As a screen reader user, I expect to have a meaningful label read aloud for every form element on the page.

## Details
When I select Other Location and enter a valid zip code, the checkbox label "I'll be taking classes outside of the U.S. and U.S. territories" is showing an axe-core critical violation. This means I won't be able to hear the label read when the checkbox receives focus. Screenshot attached below.

## Acceptance Criteria

- [ ] Zero axe-core violations are shown on future scans
- [ ] Input label is read aloud when the checkbox has screen reader focus

## Definition of done

1. Review and acknowledge feedback
1. Fix the issue documented
1. Verify 2-3 screen readers are reading out the label corectly
1. Consider adding an e2e test with axe check for this use case

## Environment

* https://staging.va.gov/gi-bill-comparison-tool/profile/10000008
* Any screen reader, any OS

## Steps to Recreate

1. Enter `https://staging.va.gov/gi-bill-comparison-tool/profile/10000008` in browser
1. Start screenreading device listed in Environment
1. Expand Learning format and schedule accordion
1. Click "Other location" for where you will take the majority of your classes
1. Enter a valid zip code
1. Press `TAB` to set focus on the checkbox. Verify no label is read aloud.

## Solution (if known)
Adjust the checkbox ID to match the `for=<UNIQUE_STRING>`

```diff
<div class="form-checkbox">
  <input
- id="classesOutsideUS"
+ id="errorable-checkbox-15"
    name="classesOutsideUS"
    type="checkbox"
  >
  <label
    class="gi-checkbox-label"
!   for="errorable-checkbox-15"
  >
    I'll be taking classes outside of the U.S. and U.S. territories
  </label>
</div>
```

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)
* [Name, Role, Value:
Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)

## Screenshots or Trace Logs

![Screen Shot 2020-10-05 at 1 24 35 PM](https://user-images.githubusercontent.com/934879/95118392-dab30000-070f-11eb-8308-d3fab88d69a9.png)

**Comment by elder-joseph** [comment]
Associated PR: https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets-website/14493

**Comment by delli-gatti-michael** [comment]
Local testing complete using the following:
- Voiceover + Chrome
- JAWS + IE
- NVDA + IE
- Axe

**Comment by delli-gatti-michael** [comment]
@1Copenut This fix is now on staging

**Comment by 1Copenut** [comment]
@delli-gatti-michael Fix looks good. I'll move this to closed.
